packages/wrangler/src/api/form_data.ts

```diff
@@ -27,9 +28,23 @@ function toBinding(
     return { name, type: "plain_text", text: variable };
   }
 
-  const { namespaceId } = variable as CfKvNamespace;
-  if (namespaceId) {
-    return { name, type: "kv_namespace", namespace_id: namespaceId };
+  if ((variable as CfKvNamespace).namespaceId) {
```

Comment:
There is an opportunity to use type guards here and below to avoid having to keep casting:

```ts
function isKvNamespace(variable: CfVariable): variable is CfKvNamespace {
  return (variable as CfKvNamespace).namespaceId;
}
```

etc.